src/main/java/mate/academy/dao/impl/CarDaoImplJdbc.java

```diff
@@ -0,0 +1,206 @@
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            removeDrivers(car, connection);
+            insertDrivers(car, connection);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't delete car by ID "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars_drivers.driver_id = ? "
+                + "AND cars.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseDataFromResultSet(resultSet));
+            }
+            setCars(cars, connection);
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get car drivers by ID "
+                    + driverId, e);
+        }
+
+    }
+
+    private void setCars(List<Car> cars, Connection connection) {
+        for (Car car : cars) {
+            car.setDrivers(getCarDrivers(car.getId(), connection));
+        }
+    }
+
+    private Car parseDataFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(name, country);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(model, manufacturer);
+        car.setId(id);
+        return car;
+    }
+
+    private List<Driver> getCarDrivers(Long id, Connection connection) {
```

Comment:
depending on the approach that you choose for `get()`, `getAll()` and `getAllByDriver()` methods - you can open a new connection in this method if you call it after the previous connection is closed

Reply:
Ok.